ProcessLib/VectorMatrixAssembler.cpp

```diff
@@ -57,8 +58,18 @@ void VectorMatrixAssembler::assemble(
     }
     else
     {
-        auto local_coupled_xs0 = getPreviousLocalSolutions(*cpl_xs, indices);
-        auto local_coupled_xs = getCurrentLocalSolutions(*cpl_xs, indices);
+        std::vector<std::reference_wrapper<const std::vector<GlobalIndexType>>>
+            indices_of_all_coupled_processes;
+        indices_of_all_coupled_processes.reserve(cpl_xs->coupled_xs.size());
+        for (std::size_t i = 0; i < cpl_xs->coupled_xs.size(); i++)
+        {
+            indices_of_all_coupled_processes.emplace_back(std::ref(indices));
+        }
+
+        auto local_coupled_xs0 = getPreviousLocalSolutions(
+            *cpl_xs, indices_of_all_coupled_processes);
+        auto local_coupled_xs =
+            getCurrentLocalSolutions(*cpl_xs, indices_of_all_coupled_processes);
```

Comment:
I'd think it is possible to pass `indices` instead of `indices_of_all_coupled_processes` because all of the entries do contain the same `indices` just several times. In the `getCurrentLocalSolutions` it could be removed also... Maybe I'm missing something.

Reply:
In VectorMatrixAssembler.cpp, the part about fetching indices has been simplified, and it is now more general for coupling and easy to understand.